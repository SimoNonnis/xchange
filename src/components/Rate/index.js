import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  selectedCode: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired
};

const Rate = ({ selectedCode, name, amount }) => (
  <div className="rate">
    <span>1 {selectedCode}</span>
    <div>
      <span className="u-block">{amount}</span>
      <span className="u-block">{name}</span>
    </div>
  </div>
);

Rate.propTypes = propTypes;

export default Rate;
