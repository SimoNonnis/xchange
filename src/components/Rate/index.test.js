import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Rate from './index';

describe('Test <Rate />', () => {
  it('should render with required props', () => {
    const wrapper = shallow(<Rate selectedCode="RUB" name="Euro" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with all props', () => {
    const wrapper = shallow(
      <Rate selectedCode="RUB" amount={1000} name="Euro" />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
