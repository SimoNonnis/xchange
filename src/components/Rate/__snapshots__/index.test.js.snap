// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Rate /> should render with all props 1`] = `
<div
  className="rate"
>
  <span>
    1 
    RUB
  </span>
  <div>
    <span
      className="u-block"
    >
      1000
    </span>
    <span
      className="u-block"
    >
      Euro
    </span>
  </div>
</div>
`;

exports[`Test <Rate /> should render with required props 1`] = `
<div
  className="rate"
>
  <span>
    1 
    RUB
  </span>
  <div>
    <span
      className="u-block"
    >
      0
    </span>
    <span
      className="u-block"
    >
      Euro
    </span>
  </div>
</div>
`;
